zebra-state/src/service/finalized_state.rs

```diff
@@ -281,7 +317,6 @@ impl FinalizedState {
         // would be harder to implement.
         //
         // TODO: run this CPU-intensive cryptography in a parallel rayon thread, if it shows up in profiles
-        let history_tree = self.db.history_tree();
         check::block_commitment_is_valid_for_chain_history(
             finalized.block.clone(),
```

Comment:
I think what's happening is that the history tree has already been updated in the non-finalized state, so it is now valid for the next block.

Previously, the finalized state validated the history tree, then updated it.
Now, we're validating it, updating it, and trying to validate it again.

As a quick fix, we can move `check::block_commitment_is_valid_for_chain_history()` to the start of the `None` block above.

(And update the comment to explain why!)

Reply:
Done in [50518fb](https://github.com/ZcashFoundation/zebra/pull/4721/commits/50518fb15c56be25b9d38c189a9f5fd91367ad82).